src/test/java/org/opensearch/flowframework/workflow/UpdateIndexStepTests.java

```diff
@@ -136,6 +200,55 @@ public void testMissingSettings() throws InterruptedException {
         );
     }
 
+    public void testUpdateMixedSettings() throws InterruptedException {
+        UpdateIndexStep updateIndexStep = new UpdateIndexStep(client);
+
+        String indexName = "test-index";
+
+        // Create existing settings for default pipelines
+        Settings.Builder builder = Settings.builder();
+        builder.put("index.number_of_shards", 2);
+        builder.put("index.number_of_replicas", 1);
+        builder.put("index.knn", true);
+        builder.put("index.default_pipeline", "ingest_pipeline_id");
+        Map<String, Settings> indexToSettings = new HashMap<>();
+        indexToSettings.put(indexName, builder.build());
+
+        // Stub get index settings request/response
+        doAnswer(invocation -> {
+            ActionListener<GetSettingsResponse> getSettingsResponseListener = invocation.getArgument(1);
+            getSettingsResponseListener.onResponse(new GetSettingsResponse(indexToSettings, indexToSettings));
+            return null;
+        }).when(indicesAdminClient).getSettings(any(), any());
+
+        // validate update settings request content
+        @SuppressWarnings({ "unchecked" })
+        ArgumentCaptor<UpdateSettingsRequest> updateSettingsRequestCaptor = ArgumentCaptor.forClass(UpdateSettingsRequest.class);
+
+        // Configurations has updated ingest pipeline default values of _none. Settings have regular and full names
+        String configurations =
```

Comment:
(Readability) This works, but in other tests I've seen us modify the first string with a regex search/replace.  Looking at this vs. line 154 I can't really visually tell what changed. I have to trust your comment (which is good that you included it, btw!)